### ISSUE TITLE
Enhancement: [no-magic-numbers] Allow for `readonly number[]`

### ISSUE THREAD

**Issue by Samuel-Therrien-Beslogic** [typescript-eslint/typescript-eslint#8102]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#supported-rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/no-magic-numbers/

### Description

Thoughts on allowing (can be locked behind an option) the following:

### Fail

```TypeScript
const DEFAULT_PAGE_LENGTH = [5, 10, 25, 100]

class MyCLass {
  readonly DEFAULT_PAGE_LENGTH = [5, 10, 25, 100]
}

let DEFAULT_PAGE_LENGTH = [5, 10, 25, 100] as const

class MyCLass {
  DEFAULT_PAGE_LENGTH = [5, 10, 25, 100] as const
}
```

### Pass

```TypeScript
const DEFAULT_PAGE_LENGTH = [5, 10, 25, 100] as const

class MyCLass {
  readonly DEFAULT_PAGE_LENGTH = [5, 10, 25, 100] as const
}
```

### Additional Info

The alternative to a eslint-disable comment really isn't preferable and really verbose:
```ts
const STEP_1 = 5
const STEP_2 = 10
const STEP_3 = 25
const STEP_4 = 10
const DEFAULT_PAGE_LENGTH = [STEP_1 , STEP_2 , STEP_3 , STEP_4]
```

**Comment by bradzacher** [comment]
As this is a stylistic extension rule - we aren't looking to deviate from the base rules enforced style. Instead the goal is to augment the existing style with support for TS features that align with the style.

The base rule has no option to ignore arrays - so we shall not add one to ingore arrays - readonly or otherwise.

**Comment by Samuel-Therrien-Beslogic** [comment]
Did not realize it was considered a stylistic rule (which I know are "no longer developed"). Thanks for your answer.

**Comment by bradzacher** [comment]
It's a stylistic rule - which you're right are considered feature complete and thus feature frozen by both us an eslint core.

But it's not considered a formatting rule - hence it has not been deprecated/removed from eslint core (and thus we retain our extension version).
